imod_coupler/drivers/dfm_metamod/dfm_metamod.py

```diff
@@ -218,8 +369,56 @@ def exchange_V_dash_1D(self) -> None:
         """
         pass
 
+    def exchange_msw2mod(self) -> None:
+        """
+        Exchange from Metaswap to MF6
+
+        1- Change of storage-coefficient from MetaSWAP to MF6
+        2- Recharge from MetaSWAP to MF6
+        3- Sprinkling request from MetaSWAP to MF6
+
+        """
+        self.mf6_storage[:] = (
+            self.mask_msw2mod["storage"][:] * self.mf6_storage[:]
+            + self.map_msw2mod["storage"].dot(self.msw_storage)[:]
+        )
+
+        # Divide recharge and extraction by delta time
+        tled = 1 / self.delt
+        self.mf6_recharge[:] = (
+            self.mask_msw2mod["recharge"][:] * self.mf6_recharge[:]
+            + tled * self.map_msw2mod["recharge"].dot(self.msw_volume)[:]
+        )
+
+        if self.coupling.enable_sprinkling:
+            self.mf6_sprinkling_wells[:] = (
+                self.mask_msw2mod["sprinkling"][:] * self.mf6_sprinkling_wells[:]
+                + tled * self.map_msw2mod["sprinkling"].dot(self.msw_volume)[:]
+            )
+
+    def exchange_mod2msw(self) -> None:
+        """
+        Exchange from MF6 to Metaswap
+
+        1- Exchange of head from MF6 to MetaSWAP
+        """
+        self.msw_head[:] = (
+            self.mask_mod2msw["head"][:] * self.msw_head[:]
+            + self.map_mod2msw["head"].dot(self.mf6_head)[:]
+        )
+
     def report_timing_totals(self) -> None:
         total_mf6 = self.mf6.report_timing_totals()
         total_msw = self.msw.report_timing_totals()
         total = total_mf6 + total_msw
         logger.info(f"Total elapsed time in numerical kernels: {total:0.4f} seconds")
+
+    def do_iter(self, sol_id: int) -> bool:
```

Comment:
I'm not sure what sol_id is, but it is called with the value "1" every time, so could it be a local variable instead of a function parameter?

Reply:
yes, i was asking myself the same question when copying it. If you look at the underlying solve in the xmiwrapper.py, it seems that it is actually a local variable right   --> solve(self, component_id=1)